src/catchup/ApplyBufferedLedgersWork.cpp

```diff
@@ -50,10 +50,18 @@ ApplyBufferedLedgersWork::onRun()
     // check for a gap
     if (lcd.getLedgerSeq() != expectedLedger)
     {
-        cm.logAndUpdateCatchupStatus(false);
-        CLOG_WARNING(History, "Expected buffered ledger={}, actual={}",
-                     expectedLedger, lcd.getLedgerSeq());
-        return State::WORK_FAILURE;
+        // The gap exists likely because of one of the following:
+        // 1. We simply didn't hear a certain ledger.
+        // 2. mSyncingLedgers has been trimmed before this work starts.
+        // 3. mSyncingLedgers was trimmed while this work is running.
+        //
+        // In any case, we will run the second catchup work.
+        // We will return SUCCESS here as ApplyBufferedLedgersWork is
+        // responsible for applying as many ledgers in mSyncingLedgers as
+        // possible.
+        CLOG_INFO(History, "Expected buffered ledger={}, actual={}",
+                  expectedLedger, lcd.getLedgerSeq());
+        return State::WORK_SUCCESS;
```

Comment:
I wasn’t quite sure how important it is to detect an anomaly. In each case, we just re-trigger catchup, so I figured that I’d put those two cases together.

Reply:
> I would suggest hiding the methods above from the main contract and do something else for `ApplyBufferedLedgersWork`.

I realized that this may be too annoying to do; I am open to alternatives - like replacing with different methods/renaming so that the use cases are clearer (unlike now).